parachain/primitives/router/src/inbound/mod.rs

```diff
@@ -83,24 +83,27 @@ pub enum Destination {
 	},
 }
 
-pub struct MessageToXcm<
-	CreateAssetCall,
-	CreateAssetDeposit,
-	InboundQueuePalletInstance,
-	AccountId,
-	Balance,
-> where
-	CreateAssetCall: Get<CallIndex>,
-	CreateAssetDeposit: Get<u128>,
+/// CallInfo required to create the foreign asset
+#[derive(Clone, Encode, Decode, RuntimeDebug)]
+pub struct CreateAssetCallInfo {
+	/// Index of the create call
+	pub call_index: CallIndex,
+	/// Deposit required to create the asset
+	pub asset_deposit: u128,
+	/// Minimal balance of the asset
+	pub min_balance: u128,
+	/// Weight required for the xcm transact
+	pub transact_weight_at_most: Weight,
+}
```

Comment:
So we add a `CallInfo` structure for the create asset and make it runtime config(without any hard coded params anymore), meanwhile we add runtime tests in asset-hub to check the `call_index` and `transact_weight_at_most` all as expected.